department-of-veterans-affairs/caseflow
# Calvin/appeals 19832

Resolves [APPEALS-19832](https://jira.devops.va.gov/browse/APPEALS-19832)

# Description

- Added new conditionals to hide legacy tasks from queue when there was active amaTasks
- Added conditionals to make distribute to judge & SpecialCaseMovementTask creation available to legacy appeals
- Task Actions on a legacy appeal now update/create both an amaTask and Legacy task in order to keep vacols up to date while having an amaTask represent it in caseflow

Workflow: With Special Case Movement, the Board is able to move an appeal from the distribution pool to a VLJ for expedited processing (No blocking tasks).

Value Statement: As a developer, I need to add functionality to the "Assign" button so that the "Case Movement" modal properly works

## Acceptance Criteria

- [ ] When Assign button is clicked, caseflow and vacols DB will be updated to complete the selected task action of "Case Movement: Advance to Judge

- [ ] priorloc and brieff VACOLS tables will be updated with re-assignment 

- [ ] Add a new row in the priorloc table when a location change is made on a Legacy Appeal

## Testing Plan
1. Go to caseflow and select user BVASYELLOW
2. Go to queue and select any of the legacy cases
3. Once in the case details, copy the URL. You will see an actions dropdown next to schedule hearing. Click the action Withdraw Hearing and continue following steps
4. Once the hearing is withdrawn, switch to a SCM User (BVASDUNKLE) and use the URL to return to the case details page
5. Once in the case details, you will see an actions dropdown next to a distribution task. Select the action Case Movement: Advance To Judge
6. Select a Judge to distribute to + instructions/reasoning then press the assign button
7. Once distributed, return to the case details page. You should see one task, an assign task assigned to the judge you selected.
8. if you go to the explain page of the case (go to url and replace "queue" with "explain"), click the checkbox task tree and you should see both a JudgeAssignTask and a JudgeLegacyAssignTask with correct user assigned to + any priorloc rows


# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?
